core/server/worker/src/main/java/alluxio/worker/page/PagedBlockStore.java

```diff
@@ -93,44 +116,89 @@ public static PagedBlockStore create(UfsManager ufsManager) {
    * @param pageMetaStore meta data store for pages and blocks
    * @param conf alluxio configurations
    */
-  PagedBlockStore(CacheManager cacheManager, UfsManager ufsManager,
-                         PageMetaStore pageMetaStore,
-                         AlluxioConfiguration conf) {
+  PagedBlockStore(CacheManager cacheManager, UfsManager ufsManager, BlockMasterClientPool pool,
+      AtomicReference<Long> workerId, PagedBlockMetaStore pageMetaStore,
+      AlluxioConfiguration conf) {
     mCacheManager = cacheManager;
     mUfsManager = ufsManager;
+    mBlockMasterClientPool = pool;
+    mWorkerId = workerId;
     mPageMetaStore = pageMetaStore;
     mConf = conf;
     mPageSize = conf.getBytes(PropertyKey.USER_CLIENT_CACHE_PAGE_SIZE);
   }
 
   @Override
   public OptionalLong pinBlock(long sessionId, long blockId) {
-    return null;
+    LOG.debug("pinBlock: sessionId={}, blockId={}", sessionId, blockId);
+    long lockId = mLockManager.lockBlock(sessionId, blockId, BlockLockType.READ);
+    if (hasBlockMeta(blockId)) {
+      return OptionalLong.of(lockId);
+    }
+    mLockManager.unlockBlock(lockId);
+    return OptionalLong.empty();
   }
 
   @Override
   public void unpinBlock(long id) {
-    throw new UnsupportedOperationException();
+    LOG.debug("unpinBlock: id={}", id);
+    mLockManager.unlockBlock(id);
   }
 
   @Override
   public void commitBlock(long sessionId, long blockId, boolean pinOnCreate) {
-    // TODO(bowen): implement actual committing and replace placeholder values
-    BlockStoreLocation dummyLoc = new BlockStoreLocation(DEFAULT_TIER, 1);
+    LOG.debug("commitBlock: sessionId={}, blockId={}, pinOnCreate={}",
+        sessionId, blockId, pinOnCreate);
+    long lockId = mLockManager.lockBlock(sessionId, blockId, BlockLockType.WRITE);
+
+    PagedBlockMeta blockMeta = mPageMetaStore.getBlock(blockId)
+        .orElseThrow(() -> new BlockDoesNotExistRuntimeException(blockId));
+    PagedBlockStoreDir pageStoreDir = blockMeta.getDir();
+    // todo(bowen): need to pin this block until commit is complete
+    // unconditionally pin this block until committing is done
```

Comment:
I'm ok with the current implementation, but I'm looking forward to your new design of the lock and pin